std/random.d

```diff
@@ -1299,19 +1373,59 @@ random number sequences every run.
 Returns:
 A single unsigned integer seed value, different on each successive call
 */
-@property uint unpredictableSeed() @trusted nothrow @nogc
+alias unpredictableSeed = unpredictableSeedOf!uint;
+/// ditto
+@property UIntType unpredictableSeedOf(UIntType)() @nogc nothrow @trusted
+if (isUnsigned!UIntType)
 {
-    import core.thread : Thread, getpid, MonoTime;
-    static bool seeded;
-    static MinstdRand0 rand;
-    if (!seeded)
+    version (AnyARC4Random)
+    {
+        // On macOS if we just need 32 bits it is faster to use
+        // arc4random() than arc4random_buf(&rslt, rslt.sizeof).
+        static if (UIntType.sizeof <= uint.sizeof)
+            return cast(UIntType) arc4random();
+        else
+        {
+            UIntType rslt = void;
+            arc4random_buf(&rslt, rslt.sizeof);
+            return rslt;
+        }
+    }
+    else static if (UIntType.sizeof <= ulong.sizeof)
+    {
+        ulong x = _seeder;
+        if (x == 0) // 0 means uninitialized (update can never transition non-zero to zero).
+            x = bootstrapSeed();
+        // State is updated with a 64-bit xorshift. Output is produced
+        // by multiplying state by a constant to eliminate linear artifacts
+        // except in the low-order bits. This approach appears in "Numerical
+        // Recipes 3rd edition" (2007). The specific constants used here are
+        // taken from "An experimental exploration of Marsaglia’s xorshift
+        // generators, scrambled" (Vigna 2016).
+        x ^= x >>> 12;
+        x ^= x << 25;
+        x ^= x >>> 37;
```

Comment:
Can't we use the existing [`Xorshift`](https://dlang.org/phobos/std_random.html#.Xorshift) range?

Reply:
I would like to but I can't. It is broken for anything but 32-bit xorshift.

Reply:
I mean xorshift with 32-bit words, although it supports several sizes of that.

Reply:
Hmm, that sucks. You are referring to https://issues.dlang.org/show_bug.cgi?id=18327, right?